src/jit/codegenxarch.cpp

```diff
@@ -2651,9 +2638,14 @@ CodeGen::genStoreRegisterReturnInLclVar(GenTreePtr treeNode)
         getStructReturnRegisters(type0, type1, &retReg0, &retReg1);
 
         assert(retReg0 != REG_NA && retReg1 != REG_NA);
-
+        
+        // Copy of a struct to a well typed location on the stack. No need to call the 
+        // GC-ness of each slot. Need to make the store of the two slots
+        // atomic for the purposes of GC.
+        getEmitter()->emitDisableGC();
```

Comment:
There should be no need to disable gc for storing the return value.  If the codegen of GT_CALL is properly reporting gc-ness of its return registers.